### PR TITLE
Feature/google search box

### PR DESCRIPTION
**Pull request by imsauravsingh**
Working on all test cases.

### CODE REVIEW

File: lib/components/molecules/Search/Search.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+import { useSpeechRecognition } from '../../../core/hooks';
+import Input from '../../atoms/Input';
+import Button from '../../atoms/Button';
+import SearchList from './SearchList/SearchList';
+import styles from './Search.style';
+import type { Props } from './types';
+
+const Search = ({ className, inputProps, dataLists, renderListItem }: Props): Node => {
+  const [inputValue, setInputValue] = useState(inputProps.value || '');
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: inputValue --> searchString looks more contextual

File: lib/components/molecules/Search/Search.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  const [searchLists, setSearchList] = useState([]);
+
+  const onSearchClick = useCallback(
+    searchText => {
+      const searchValue = typeof searchText === 'string' ? searchText : inputValue;
+
+      if (searchValue) {
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js:26]: this logic I believe would come from TypeAhead

File: lib/components/molecules/Search/Search.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          return name.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1;
+        });
+        setSearchList(lists);
+      }
+      return false;
+    },
+    [inputValue, dataLists, setSearchList]
+  );
+
+  const onResult = result => {
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: Can we make onResult more explicit? its ambiguous what it does 

onSpeachRecognitionResults makes it more clear

File: lib/components/molecules/Search/Search.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  };
+
+  const handleSpeechEnd = () => {};
+
+  const { isListening, onListen } = useSpeechRecognition({
+    onResult,
+    handleSpeechEnd,
+  });
+
+  const onSpeechClick = () => onListen();
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: onListen . --> onSpeechListenHandler

File: lib/components/molecules/Search/Search.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+          className={classnames('search-txt', inputProps.className)}
+          value={inputValue}
+          onChange={onChangeHandler}
+          onKeyPress={keyPress}
+        />
+
+        <div className="search-button">
+          <Button>
+            {isListening ? (
+              <i className="search-icon search-loader" />
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: Use Icon component

File: lib/components/molecules/Search/Search.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    onSearchClick(result);
+    return true;
+  };
+
+  return (
+    <div className={classnames(styles.default, className)}>
+      <div className="search-box">
+        <Input
+          {...inputProps}
+          aria-label="Search"
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: Labels should never be hard coded.

File: lib/components/molecules/Search/Search.style.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    outline: none;
+    flex-grow: 1;
+  }
+
+  .search-button {
+    display: flex;
+    overflow: hidden;
+  }
+
+  button {
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.style.js:28]: This is wrong don't style the component here

File: lib/components/molecules/Search/Search.style.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  .search-icon {
+    ${props => `
+    width: ${props.theme.spacing.UNIT_3};
+    height: ${props.theme.spacing.UNIT_3};
+    `};
+    display: block;
+  }
+
+  .search-mic {
+    background: url(images/googlemic.png) 0 0/24px no-repeat;
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.style.js:47]: this is not a good approach

Comment (imsauravsingh) [reply lib/components/molecules/Search/Search.style.js:47]: It's a part of the icon component. Once Icon component will be exposed then these codes will be removed.

File: lib/components/molecules/Search/Search.style.js (hunk 7 above)
Comment (imsauravsingh) [reply lib/components/molecules/Search/Search.style.js:28]: Once Icon componet will be exposed then unnecessary codes will be removed.

File: lib/components/molecules/Search/Search.js (hunk 6 above)
Comment (imsauravsingh) [reply lib/components/molecules/Search/Search.js]: yes, I'll update

File: lib/components/molecules/Search/Search.js (hunk 5 above)
Comment (imsauravsingh) [reply lib/components/molecules/Search/Search.js]: yes

**Pull request closed without merging**